src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -256,17 +284,18 @@ public virtual void WriteResponseHeaders(OutputFormatterWriteContext context)
         protected string GetMediaTypeWithCharset(string mediaType, Encoding encoding)
         {
             var mediaTypeEncoding = MediaTypeEncoding.GetEncoding(mediaType);
-            if (mediaTypeEncoding == encoding)
+            if (mediaTypeEncoding != null &&
+                mediaTypeEncoding.WebName.Equals(encoding.WebName, StringComparison.OrdinalIgnoreCase))
```

Comment:
This is the case where the accept header has a charset is it not? I don't see a test for it. I see a test for the UTF-8 cache optimization.